File: Algorithms/C++/00392-Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.size() == 0) return true;
        int sp{0};

        for(int tp{0}; tp < t.size(); tp++){
            if(s[sp] == t[tp]){
                sp++;
            }
            if(sp == s.size()) return true;
        }
        return false;
    }
};

/*
    tc: O(n)
    sc: O(1)
    2 pointers
    - iterate through t
    - if we see that t == s
    - then increment s pointer
    - return true if s reaches to the end
    - otherwise return false


*/

File: Algorithms/Python3/00392-Is Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0:
            return True
            
        sp = 0
        tp = 0

        for i in range(len(t)):
            if s[sp] == t[tp]:
                sp += 1
            if sp == len(s):
                return True
            tp += 1
        
        return False

'''
    s = abc
    t = ahbgdc

    return true if we are at end of s
    return false if we are at end of t


'''